specs/Freeze.md

```diff
@@ -0,0 +1,124 @@
+    webView.Visibility = false;
+    await webView.CoreWebView2.TryFreezeAsync();
+    deferral.Complete();
+}
+async protected void OnSuspending(object sender, Object args)
+{
+    webView.CoreWebView2.Unfreeze();
+    webView.Visibility = true;
+}
+```
+## Win32 C++
+As unfreeze is very fast and automatically happens when WebView becomes visible, the app can generaly immediately call `TryFreeze` when a WebView becomes invisible.
+```cpp
+bool ViewComponent::HandleWindowMessage(
+    HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam, LRESULT* result)
+{
+    if (message == WM_SYSCOMMAND)
+    {
+        if (wParam == SC_MINIMIZE)
+        {
+            // Hide the webview when the app window is minimized, and freeze it.
+            m_controller->put_IsVisible(FALSE);
+            Freeze();
+        }
+        else if (wParam == SC_RESTORE)
+        {
+            // When the app window is restored, show the webview
+            // (unless the user has toggle visibility off).
+            if (m_isVisible)
+            {
+                Unfreeze();
+                m_controller->put_IsVisible(TRUE);
+            }
+        }
+    }
+}
+
+void ViewComponent::Freeze()
+{
+    HRESULT hr = webView->TryFreeze(
+        Callback<ICoreWebView2StagingTryFreezeCompletedHandler>(
+            [](HRESULT errorCode, BOOL isSuccessful) -> HRESULT {
+                std::wstringstream formattedMessage;
+                formattedMessage << "TryFreeze result (0x" << std::hex << errorCode
+                                 << ") " << (isSuccessful ? "succeeded" : "failed");
+                MessageBox(nullptr, formattedMessage.str().c_str(), nullptr, MB_OK);
+                return S_OK;
+            })
+            .Get());
+    if (FAILED(hr))
+        ShowFailure(hr, L"Call to TryFreeze failed");
+}
+
+void ViewComponent::Unfreeze()
+{
+    wil::com_ptr<ICoreWebView2Staging2> webView;
+    webView = m_webView.query<ICoreWebView2Staging2>();
+    webView->Unfreeze();
+}
+```
+
+# Remarks
+WebView2 controller must be invisible when the API is called. Otherwise, the
+API fails with `HRESULT_FROM_WIN32(ERROR_INVALID_STATE)`.   
+Freezing is similar to putting a tab into sleeping in browser. Freezing pauses
+WebView script timers and animations, minimizes CPU usage for the associated
+browser renderer process and allows the operating system to reuse the memory that was
+used by the renderer process for other processes.   
+See [Sleeping Tabs FAQ](https://techcommunity.microsoft.com/t5/articles/sleeping-tabs-faq/m-p/1705434)
+for conditions that might prevent WebView from being frozen. In those situations,
+TryFreeze operation will fail and the completed handler will be invoked with isSuccessful as false.   
+WebView will be automatically unfrozen when it becomes visible. Therefore, the app normally doesn't have to call Unfreeze.
+The app can call `Unfreeze` and then `TryFreeze` periodically for an invisibile WebView so that the WebView can sync up with
```

Comment:
Is the implication here that after calling Unfreeze the WebView will be visible but will look weird or wrong with visual artifacts or flashing or something? Or just that once unfrozen, timers in script will start firing again and if those do anything to update the page you want to let them run earlier to ensure once visible the WebView2 looks good?

Reply:
The WebView will stay as invisible. Will change to
so that the invisible WebView can sync up with
latest data and the page ready to show fresh content when it becomes visible.